sched/task/task_vfork.c

```diff
@@ -151,6 +151,14 @@ FAR struct task_tcb_s *nxtask_setup_vfork(start_t retaddr)
       goto errout_with_tcb;
     }
 
+  /* Duplicate the parent tasks environment */
+
+  ret = env_dup(child);
```

Comment:
This is correct, `fork()` and `vfork()` implicitly copy the parent's environment as is to the child. The child is free to continue as is, or call `execve()` / similar with a new envp which will replace the running process and thus replace the copied env.